Makefile

```diff
@@ -0,0 +1,63 @@
+# Makefile for Harbor-arm project
+#
+# Targets:
+#
+# all: 
+
+BUILDPATH=$(CURDIR)
+HARBOR_MAKEFILE_PATH=$(BUILDPATH)/Makefile
+HARBOR_PHOTON_MAKEFILE_PATH=$(BUILDPATH)/make/photon/Makefile
+HARBOR_PHOTON_CHARTSERVER_COMPILE_PATH=$(BUILDPATH)/make/photon/chartserver/compile.sh
+HARBOR_PHOTON_REGISTRY_DOCKERFILE_PATH=$(BUILDPATH)/make/photon/registry/Dockerfile.binary
+HARBOR_PHOTON_TRIVY-ADAPTER_DOCKERFILE_PATH=$(BUILDPATH)/make/photon/trivy-adapter/Dockerfile.binary
+HARBOR_PHOTON_NOTARY_DOCKERFILE_PATH=$(BUILDPATH)/make/photon/notary/binary.Dockerfile
+HARBOR_PHOTON_PORTAL_DOCKERFILE_PATH=$(BUILDPATH)/make/photon/portal/Dockerfile
+HARBOR_PUSHIMAGE_PATH=$(BUILDPATH)/make/pushimage.sh
+
+SEDCMD=$(shell which sed)
+SEDCMDI=$(SEDCMD) -i
+ifeq ($(shell uname),Darwin)
+    SEDCMDI=$(SEDCMD) -i ''
+endif
+
+_update_makefile:
+	@echo "update goharbor makefile"
+	@$(SEDCMDI) 's/--rm/--rm --env CGO_ENABLED=0 --env GOOS=linux --env GOARCH=arm64/g' $(HARBOR_MAKEFILE_PATH);
+	@$(SEDCMDI) 's/$$(DOCKERBUILD)/docker buildx build --platform linux\/arm64 --progress plain --output=type=docker/g' $(HARBOR_MAKEFILE_PATH)
+
+_update_make_photon_makefile:
+	@echo "update goharbor photon makefile"
+	@$(SEDCMDI) 's/$(DOCKERCMD) build/$(DOCKERCMD) buildx build --platform linux\/arm64 --progress plain --output=type=docker/' $(HARBOR_PHOTON_MAKEFILE_PATH)
+
+_update_chartserver:
+	@echo "update goharbor chartserver compile.sh"
+	@$(SEDCMDI) 's/go build -a/GOOS=linux GOARCH=arm64 CGO_ENABLED=0 go build -a/g' $(HARBOR_PHOTON_CHARTSERVER_COMPILE_PATH)
+
+_update_registry:
+	@echo "update goharbor Dockerfile.binary"
+	@$(SEDCMDI) 's/CGO_ENABLED=0/GOOS=linux GOARCH=arm64 CGO_ENABLED=0/g' $(HARBOR_PHOTON_REGISTRY_DOCKERFILE_PATH)
+
+_update_trivy-adapter:
```

Comment:
Build trivy adapter we also need to replace the trivy binary url, because default trivy binary is x86 format. ref: https://github.com/goharbor/harbor/blob/b789674ada2ed21fb5277daeb55657c20788de6b/Makefile#L129

Reply:
your mean's we need to update TRIVY_ADAPTER_DOWNLOAD_URL?

Reply:
TRIVY_DOWNLOAD_URL

Reply:
can't replacable, upstream not provider aarch64 version

have to rebuild theme in Dockerfile  https://github.com/morlay/harbor/blob/07c17def38254b4abcb06528d191ccf16014363a/make/photon/trivy-adapter/Dockerfile.binary

Reply:
I mean trivy, not trivy-adapter, https://github.com/aquasecurity/trivy/releases.

Reply:
ok，i'll update it；
There are big changes regarding the build replacement, I will submit the local changes as soon as possible